equinix/resource_fabric_network.go

```diff
@@ -0,0 +1,431 @@
+		mappedChanges,
+	)
+	return changeSet
+}
+
+func setFabricNetworkMap(d *schema.ResourceData, nt v4.Network) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+	err := equinix_schema.SetMap(d, map[string]interface{}{
+		"name":              nt.Name,
+		"href":              nt.Href,
+		"uuid":              nt.Uuid,
+		"type":              nt.Type_,
+		"scope":             nt.Scope,
+		"state":             nt.State,
+		"operation":         fabricNetworkOperationToTerra(nt.Operation),
+		"change":            simplifiedFabricNetworkChangeToTerra(nt.Change),
+		"location":          equinix_fabric_schema.LocationToTerra(nt.Location),
+		"notifications":     equinix_fabric_schema.NotificationsToTerra(nt.Notifications),
+		"project":           equinix_fabric_schema.ProjectToTerra(nt.Project),
+		"change_log":        equinix_fabric_schema.ChangeLogToTerra(nt.ChangeLog),
+		"connections_count": nt.ConnectionsCount,
+	})
+	if err != nil {
+		return diag.FromErr(equinix_errors.FormatFabricError(err))
+	}
+	return diags
+}
+func getFabricNetworkUpdateRequest(network v4.Network, d *schema.ResourceData) (v4.NetworkChangeOperation, error) {
+	changeOps := v4.NetworkChangeOperation{}
+	existingName := network.Name
+	updateNameVal := d.Get("name")
+
+	log.Printf("existing name %s, Update Name Request %s ", existingName, updateNameVal)
+
+	if existingName != updateNameVal {
+		changeOps = v4.NetworkChangeOperation{Op: "replace", Path: "/name", Value: &updateNameVal}
+	} else {
+		return changeOps, fmt.Errorf("nothing to update for the Fabric Network: %s", existingName)
+	}
+	return changeOps, nil
+}
+func resourceFabricNetworkUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.Config).FabricClient
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, meta.(*config.Config).FabricAuthToken)
+	dbConn, err := waitUntilFabricNetworkIsProvisioned(d.Id(), meta, ctx)
+	if err != nil {
+		if !strings.Contains(err.Error(), "500") {
+			d.SetId("")
+		}
+		return diag.Errorf("either timed out or errored out while fetching Fabric Network for uuid %s and error %v", d.Id(), equinix_errors.FormatFabricError(err))
+	}
+	// TO-DO
+	update, err := getFabricNetworkUpdateRequest(dbConn, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	updates := []v4.NetworkChangeOperation{update}
+	_, res, err := client.NetworksApi.UpdateNetworkByUuid(ctx, updates, d.Id())
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error response for the Fabric Network update, response %v, error %v", res, equinix_errors.FormatFabricError(err)))
+	}
+	updateFg := v4.Network{}
+	updateFg, err = waitForFabricNetworkUpdateCompletion(d.Id(), meta, ctx)
+
+	if err != nil {
+		if !strings.Contains(err.Error(), "500") {
+			d.SetId("")
+		}
+		return diag.FromErr(fmt.Errorf("errored while waiting for successful Fabric Network update, response %v, error %v", res, equinix_errors.FormatFabricError(err)))
```

Comment:
The error handling for the wait methods will not be able to pass the error through the formatter twice. It is already handled by you in the wait methods. So outside of them we can just place it into the fmt.Errorf as is.